### PR TITLE
Added streaming to api

### PR DESCRIPTION
**Pull request by kateknister**

### CODE REVIEW

File: api/handler.go [hunk 1]
```diff
@@ -130,6 +132,38 @@ func writeResult(res interface{}, w http.ResponseWriter) error {
 
 }
 
+func writeStreamResult(results chan *events.Event, w http.ResponseWriter, r *http.Request) error {
```
Comment (vmarmol) [inline api/handler.go]: nit: rename to "streamResults"

File: api/handler.go [hunk 2]
```diff
@@ -130,6 +132,38 @@ func writeResult(res interface{}, w http.ResponseWriter) error {
...
+	cn, ok := w.(http.CloseNotifier)
+	if !ok {
+		http.NotFound(w, r)
```
Comment (vmarmol) [inline api/handler.go]: lets not use an http error here and below since the API code takes care of it

File: api/handler.go [hunk 3]
```diff
@@ -130,6 +132,38 @@ func writeResult(res interface{}, w http.ResponseWriter) error {
...
+		return errors.New("Could not access http.CloseNotifier")
```
Comment (vmarmol) [inline api/handler.go]: nit: lowercase could

File: api/handler.go [hunk 4]
```diff
@@ -130,6 +132,38 @@ func writeResult(res interface{}, w http.ResponseWriter) error {
...
+	}
+	flusher, ok := w.(http.Flusher)
+	if !ok {
+		http.NotFound(w, r)
+		return errors.New("Could not access http.Flusher")
```
Comment (vmarmol) [inline api/handler.go]: nit: lowercase could

File: api/handler.go [hunk 5]
```diff
@@ -130,6 +132,38 @@ func writeResult(res interface{}, w http.ResponseWriter) error {
...
+	enc := json.NewEncoder(w)
+	for {
+		select {
+		case <-cn.CloseNotify():
+			glog.Infof("Client stopped listening")
+			return nil
+		case ev := <-results:
+			err := enc.Encode(ev)
+			if err != nil {
+				glog.Errorf("found fatal in encoding in modify page: %v", err)
```
Comment (vmarmol) [inline api/handler.go]: Change to: "Error encoding message %+v for result stream: %v", ev, err

**Comment by vmarmol** [comment]
This looks great @kateknister! Only minor comments.

**Comment by kateknister** [comment]
Here is an updated version!

File: api/handler.go [hunk 6]
```diff
@@ -130,6 +132,36 @@ func writeResult(res interface{}, w http.ResponseWriter) error {
...
+
+	w.Header().Set("Transfer-Encoding", "chunked")
+	w.WriteHeader(http.StatusOK)
+	flusher.Flush()
+
+	enc := json.NewEncoder(w)
+	for {
+		select {
+		case <-cn.CloseNotify():
+			glog.Infof("Client stopped listening")
```
Comment (vmarmol) [inline api/handler.go]: lets remove this log

**Comment by vmarmol** [comment]
LGTM, will merge, feel free to remove the log line in the followup PR.

**Pull request merged**